evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -180,10 +178,48 @@ def categories(self, feature_name):
             raise ValueError(f'Feature "{feature_name}" was not provided to one-hot encoder as a training feature')
         return self._encoder.categories_[index]
 
+    @staticmethod
+    def _make_name_unique(name, seen_before):
+        """Helper to make the name unique."""
+
+        if name not in seen_before:
+            return name
+
+        # Only modify the name if it has been seen before
+        i = 1
+        name = f"{name}_{i}"
+        while name in seen_before:
+            name = f"{name[:name.rindex('_')]}_{i}"
+            i += 1
+        return name
+
     def get_feature_names(self):
-        """Return feature names for the input features after fitting.
+        """Return feature names for the categorical features after fitting.
+
+        Feature names are formatted as {column name}_{category name}. In the event of a duplicate name,
+        an integer will be added at the end of the feature name to distinguish it.
+
+        For example, consider a dataframe with a column called "A" and category "x_y" and another column
+        called "A_x" with "y". In this example, the feature names would be "A_x_y" and "A_x_y_1".
```

Comment:
Edge case: does this fix address the case where you have a feature "A_x" with category "y" and a feature "A" with category "x_y"?

Reply:
I think this is the first case in `test_ohe_column_names_unique` but let me know if I'm misunderstanding!

Reply:
Cool!